### PR TITLE
Drop fixed pipeline support code

### LINKED ISSUE #15370: Drop fixed pipeline support code
continuation of #15067 (in spirit), #15210

## Current situation

Shaders have been enabled by default since ~0.4.16 on desktop and 5.7.0 on Android.

## Proposal

Delete all code that depends on `!enable_shaders`.
It's not easy to estimate how much this is but `MapBlockMesh` has entire data structures for doing lighting only when shaders are not used.

_Why?_ I think it's finally time for us to free ourselves from legacy shackles.
If you pull a 10 year old computer from a dusty store room it will be able to support OpenGL with shaders. Refer also to [this discussion](https://irc.minetest.net/minetest-dev/2022-08-18).
There will inevitably be cases where Luanti runs better w/o shaders than with. Hopefully users will report them to us and we can try to fix them, but in many cases this is simply due to outdated hardware design ([example](https://irc.minetest.net/minetest-dev/2024-11-10#i_6216270)).

This would effectively put our minimum OpenGL requirement at 2.0.

## Advantages

* less and cleaner code to maintain
* less bugs that nobody notices (#15364)
* bigger potential for optimizations to be made that require shaders
  * (theoretically possible now but in many cases we don't want to maintain two implementations of something)
* future features can be added that require shaders (e.g. #14343)

## Concerns

* `video_driver = null` should be kept working for headless usage, but this is easy
* the number of negatively impacted users will be non-zero. we will have to accept this.

Note: With menu clouds relying on shaders in 5.10.0, we will be able to tell quite well from user complaints if some people really have shader-incompatible hardware.

### PR DESCRIPTION
**Pull request by sfan5**
resolves #15370

## To do

This PR is Ready for Review.

## How to test

1. check for rendering regressions

### CODE REVIEW

File: src/client/mapblock_mesh.cpp [hunk 1]
```diff
@@ -729,12 +703,10 @@ MapBlockMesh::MapBlockMesh(Client *client, MeshMakeData *data, v3s16 camera_offs
 				tex.MagFilter = video::ETMAGF_NEAREST;
 			});
 
-			if (m_enable_shaders) {
+			{
 				material.MaterialType = m_shdrsrc->getShaderInfo(
 						p.layer.shader_id).material;
 				p.layer.applyMaterialOptionsWithShaders(material);
```
Comment (lhofhansl) [inline src/client/mapblock_mesh.cpp:709]: Should we rename that function and remove the "withShaders" part?

**Review by lhofhansl — APPROVED** [review]
Works for me.
Looked through the code, all looks good.
Nice to see all this code removed.

File: src/client/mapblock_mesh.cpp (hunk 1 above)
Comment (sfan5) [reply src/client/mapblock_mesh.cpp:709]: `applyMaterialOptions` is still in use, unfortunately.

**Comment by lhofhansl** [comment]
@sfan5 Looks like this breaks the "effect" dialog:
```
2024-11-13 09:09:50: ERROR[Main]: Main menu error: Runtime error from mod '' in callback handleMainMenuButtons(): .../share/luanti/builtin/mainmenu/settings/dlg_settings.lua:362: attempt to concatenate local 'name' (a nil value)
2024-11-13 09:09:50: ERROR[Main]: stack traceback:
2024-11-13 09:09:50: ERROR[Main]:       .../share/luanti/builtin/mainmenu/settings/dlg_settings.lua:362: in function 'check_requirements'
2024-11-13 09:09:50: ERROR[Main]:       .../share/luanti/builtin/mainmenu/settings/dlg_settings.lua:433: in function 'build_page_components'
2024-11-13 09:09:50: ERROR[Main]:       .../share/luanti/builtin/mainmenu/settings/dlg_settings.lua:560: in function 'get_formspec'
2024-11-13 09:09:50: ERROR[Main]:       /opt/luanti/share/luanti/builtin/fstk/ui.lua:110: in function 'update'
2024-11-13 09:09:50: ERROR[Main]:       /opt/luanti/share/luanti/builtin/fstk/ui.lua:154: in function 'handle_buttons'
2024-11-13 09:09:50: ERROR[Main]:       /opt/luanti/share/luanti/builtin/fstk/ui.lua:195: in function </opt/luanti/share/luanti/builtin/fstk/ui.lua:181>
```

**Pull request merged**